frontend/src/composables/useTokenSelection.ts

```diff
@@ -0,0 +1,57 @@
+import { computed, Ref, ref } from 'vue';
+
+import { getTokenDecimals } from '@/services/transactions/token';
+import type { EthereumProvider } from '@/services/web3-provider';
+import { useConfiguration } from '@/stores/configuration';
+import { Token } from '@/types/config';
+import { SelectorOption } from '@/types/form';
+
+export function useTokenSelection(provider: Ref<EthereumProvider | undefined>) {
+  const configuration = useConfiguration();
```

Comment:
Again: if I'm not mistaken we decided against using stores directly in composables.